### ISSUE TITLE
Implement `pct help` subsystem

### ISSUE THREAD

**Issue by sanfrancrisko** [puppetlabs/prm#11]
Implement the `pct help` subsystem. This should provide:

- Overview of the command / subcommand
- A flag for documents by topic
- Links to online docs

Examples:

```sh
prm help --topic validating-bolt-tasks # Pulls up walkthru
prm help validate # Pulls up reference docs for prm validate
prm validate --help # Pulls up reference docs for prm validate
prm help set puppet # Pulls up reference docs for prm set puppet
prm help --topic backend # Concept doc for backends generally
prm help --topic backend-docker # Concept doc for configuring/using docker backend
prm help --tool rubocop # Documentation for the rubocop tool use via PRM, link to upstream tool docs
```

**Comment by michaeltlombardi** [comment]
I've done some investigation into custom help and cobra today with some useful findings to summarize here:

1. Command level help is automatic, so we don't actually need to do any substantial work for surfacing reference docs for commands; `prm help set puppet` will just work out of the box.
2. We _can_ write additional subcommands such as `prm exec describe puppetlabs/rubocop` - with some shared code leveraged to make the same functionality available in `validate`.
3. I looked into the cobra concept of additional help commands; unfortunately, I think implementing something (with an example [described here](https://github.com/spf13/cobra/issues/393#issuecomment-282741924)) would require metaprogramming to create these on the fly and runs the risk of clobbering the expected behavior as whether `prm exec puppetlabs/rubocop --help` should return the help as surfaced by the tool _or_ the help about the tool package is not obvious. I suggest instead that we go with the direction in point 2 above for surfacing information about the tool package.
4. Adding flags to the base help command is _possible_ but _poor UX_ - calling `prm help` displays the help for the root prm command without any additional flags for help; those flags _are_ visible if the user calls `prm help help` but this is incredibly non-obvious. So while `prm help --topic foo` is _technically implementable_ I recommend we instead add an `explain` subcommand which can read docs by topic etc.

In summary:

- using the existing help subsystem to surface reference docs for commands is functional and useful
- we can have the commands for `exec` and `validate` include a `describe` subcommand which can surface the reference docs for that tool as packaged for PRM
- we can include an `explain` subcommand for surfacing topics written in markdown and sorted by topic/etc
- we can further investigate how we want to surface the actual tool's help information (ie `rubocop --help` in the container) via exec/validate without clobbering other information. Possibly simplest to do with a simple flag on the exec and validate commands.

**Comment by michaeltlombardi** [comment]
Resolved by https://github.com/puppetlabs/prm/pull/83
